server/coordinator/procedure/manager_impl.go

```diff
@@ -13,79 +18,103 @@ import (
 )
 
 const (
-	queueSize         = 10
-	metaListBatchSize = 100
+	metaListBatchSize        = 100
+	defaultWaitingQueueSize  = 100
+	defaultWaitingQueueDelay = time.Millisecond * 500
 )
 
 type ManagerImpl struct {
-	// This lock is used to protect the field `procedures` and `running`.
-	lock       sync.RWMutex
-	procedures []Procedure
-	running    bool
+	storage        Storage
+	dispatch       eventdispatch.Dispatch
+	clusterManager cluster.Manager
 
-	storage  Storage
-	dispatch eventdispatch.Dispatch
+	// This lock is used to protect the following fields.
+	lock     sync.RWMutex
+	running  bool
+	clusters map[storage.ClusterID]*ClusterProcedures
```

Comment:
Why not create a `ClusterProcedures`(may we should call it `ProcedureManager`) for every single cluster and attach such `ClusterProcedures` directly to the cluster struct.

Reply:
Use `cluster` to manage `procedureManager` and `schedulerManager`, in order to avoid circular dependency, rename the original `cluster` package to `metadata`.